test/integration/admin/api/settings_controller_test.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class Admin::Api::SettingsControllerTest < ActionDispatch::IntegrationTest
+  def setup
+    provider = FactoryBot.create(:provider_account)
+    host! provider.external_admin_domain
+    @token = FactoryBot.create(:access_token, owner: provider.admin_user, scopes: %w[account_management]).value
+    @settings = provider.settings
+  end
+
+  attr_reader :settings, :token
+
+  test 'show' do
+    get admin_api_settings_path(access_token: token, format: :json)
+    expected_response = {
+      settings: {
+        useraccountarea_enabled: true,
+        signups_enabled: true,
+        account_approval_required: false,
+        strong_passwords_enabled: false,
+        public_search: false,
+        account_plans_ui_visible: true,
+        change_account_plan_permission: 'request',
+        service_plans_ui_visible: true,
+        change_service_plan_permission: 'request'
+      }
+    }.as_json
+    assert_response :success
+    assert_equal expected_response, JSON.parse(response.body)
+  end
```

Comment:
Not super important, but I think tests should fit the following format
```ruby
def test
  preparation
  run
  assert
end
```

So in this case:

```suggestion
  test 'show' do
    expected_response = {
      settings: {
        useraccountarea_enabled: true,
        signups_enabled: true,
        account_approval_required: false,
        strong_passwords_enabled: false,
        public_search: false,
        account_plans_ui_visible: true,
        change_account_plan_permission: 'request',
        service_plans_ui_visible: true,
        change_service_plan_permission: 'request'
      }
    }.as_json
    
    get admin_api_settings_path(access_token: token, format: :json)
    
    assert_response :success
    assert_equal expected_response, JSON.parse(response.body)
  end
```

Reply:
In general I agree with this format, it makes sense.
However, in this case defining the `expected_response` is not "preparation", it's just extracting a part of the assertion into a variable for better visibility.

It could also be just

```
    assert_response :success
    assert_equal {
      settings: {
        useraccountarea_enabled: true,
        signups_enabled: true,
        account_approval_required: false,
        strong_passwords_enabled: false,
        public_search: false,
        account_plans_ui_visible: true,
        change_account_plan_permission: 'request',
        service_plans_ui_visible: true,
        change_service_plan_permission: 'request'
      }
    }.as_json, JSON.parse(response.body)
```

But IMO it's just more confusing.

So, I'd prefer to keep the variable closer to where it is used, no need to split it by the API invocation.